src/odemis/util/img.py

```diff
@@ -591,3 +592,64 @@ def mergeMetadata(current, correction=None):
         if k in current:
             del current[k]
 
+def mergeTiles(tiles):
+    """"
+    Merge tiles into one DataArray
+    tiles (tuple of tuple of DataArray): Tiles to be merged
+    return (DataArray): Merge of all the tiles
+    """
+
+    def get_corner_pos(tile, cm):
+        md = tile.metadata
+        dims = md.get(model.MD_DIMS, "CTZYX"[-tile.ndim::])
+        tile_shape = [tile.shape[dims.index('X')], tile.shape[dims.index('Y')]]
+        corner_dist_to_center_pixels = numpy.array([cm[0] * tile_shape[0]/2, cm[1] * tile_shape[1]/2])
+        corner_dist_to_center_pixels = numpy.matrix(corner_dist_to_center_pixels).getT()
+        tmat = get_img_transformation_matrix(tiles[0][0].metadata)
+        rel_centerw = tmat * corner_dist_to_center_pixels
+        rel_centerw = numpy.ravel(rel_centerw)
+        centerw = tile.metadata[model.MD_POS]
+        return [rel_centerw[0] + centerw[0], rel_centerw[1] + centerw[1]]
+
+
+    height = 0
+    for tile in tiles[0]:
+        height += tile.shape[0]
+
+    width = 0
+    for tiles_column in tiles:
+        width += tiles_column[0].shape[1]
+
+    result_shape = (height, width)
+    # TODO must work when the channel dimension is not the last
+    if len(tiles[0][0].shape) == 3:
+        result_shape = result_shape + (tiles[0][0].shape[2],)
+
+    # TODO support multiple channels
+    result = numpy.zeros(result_shape, dtype=tiles[0][0].dtype)
+    result = model.DataArray(result, tiles[0][0].metadata.copy())
+
+    width_sum = 0
+    for tiles_column in tiles:
+        tile_width = tiles_column[0].shape[1]
+        height_sum = 0
+        for tile in tiles_column:
+            tile_height = tile.shape[0]
+            result[height_sum:(height_sum + tile_height), width_sum:(width_sum + tile_width)] = \
+                tile
+            height_sum += tile_height
+
+        width_sum += tile_width
+
+    num_rows = len(tiles)
+    num_cols = len(tiles[0])
+
+    first_tile_pos = get_corner_pos(tiles[0][0], [-1, -1])
```

Comment:
Can you explain a bit further how do you want this part to change? The way to be similar to  `get_tile_md_pos` and `get_img_transformation_matrix` is to convert pixel coordinates to world coordinates as soon as possible?
Right now, what it is doing in the end of the function, is to get the top-left and bottom-right corners of the image and calculate the medium point.
I added some more comments.